server/api/crud/runtimes/nuclio/function.py

```diff
@@ -324,6 +326,16 @@ def _compile_function_config(
         config=function.spec.config,
     )
 
+    build: mlrun.model.ImageBuilder = function.spec.build
+    base_image: str = (
+        build.base_image or function.spec.image or mlrun_config.default_base_image
+    )
+
+    if server.api.utils.builder.is_mlrun_image(base_image) and build.requirements:
+        enriched_base_image = function.full_image_path(
+            base_image, client_version, client_python_version
+        )
```

Comment:
this is duplicated with `_resolve_and_set_base_image` below. see if you can move `_resolve_and_set_build_requirements_and_commands` after `_resolve_and_set_base_image` and I would also move the added code into `_resolve_and_set_build_requirements_and_commands`

Reply:
Do you mean that this is the duplication?
```
base_image = (
        mlrun.utils.get_in(config, "spec.build.baseImage")
        or function.spec.image
        or function.spec.build.base_image
    )
```

Reply:
yes and the enrichment below